opentelemetry-etw-logs/src/logs/reentrant_logprocessor.rs

```diff
@@ -6,40 +6,29 @@ use opentelemetry_sdk::logs::{LogBatch, LogExporter, SdkLogRecord};
 use opentelemetry_sdk::Resource;
 
 use crate::logs::exporter::*;
-
-pub fn validate_provider_name(provider_name: &str) -> Result<(), String> {
-    if provider_name.is_empty() {
-        return Err("Provider name cannot be empty.".to_string());
-    }
-    if provider_name.len() >= 234 {
-        return Err("Provider name must be less than 234 characters.".to_string());
-    }
-    if !provider_name
-        .chars()
-        .all(|c| c.is_ascii_alphanumeric() || c == '_')
-    {
-        return Err("Provider name must contain only ASCII letters, digits, and '_'.".to_string());
-    }
-    Ok(())
-}
+use crate::logs::ExporterOptions;
 
 /// Thread-safe LogProcessor for exporting logs to ETW.
-
 #[derive(Debug)]
-pub struct ReentrantLogProcessor {
+pub(crate) struct ReentrantLogProcessor {
     event_exporter: ETWExporter,
 }
 
 impl ReentrantLogProcessor {
-    /// constructor
-    pub fn new(provider_name: &str) -> Self {
-        let exporter = ETWExporter::new(provider_name);
+    /// Creates a new instance of the ReentrantLogProcessor using the given options.
+    pub(crate) fn new(options: ExporterOptions) -> Self {
+        let exporter: ETWExporter = ETWExporter::new(options);
         ReentrantLogProcessor {
             event_exporter: exporter,
         }
     }
 }
 
+/// Creates an opaque LogProcessor that can be used with the OpenTelemetry SDK.
```

Comment:
given this returns a *new* instance of LogRecordProcessor, *not* the one actually used if user did `with_etw_exporter` option to add etw_exporter, this can make it easy for users to make mistakes!
(it won't be obvious that the processor returned in this method is not the same as what is used by the trait method!)

Reply:
> It should definitely be clarified better in the rustdoc, but I'd hope that that's enough. Or do you have something else in mind as for how to address this?

Doc is the minimum we should do. Or better is to avoid 2 APIs, so there is no scope of misconfiguring things.